src/controllers/csi/provisioner/controller.go

```diff
@@ -89,7 +89,7 @@ func (provisioner *OneAgentProvisioner) Reconcile(ctx context.Context, request r
 		}
 		return reconcile.Result{}, err
 	}
-	if !dk.NeedsCSIDriver() {
+	if !dk.NeedsCSIDriver() || !dk.NeedAppInjection() {
 		log.Info("CSI driver not needed")
```

Comment:
nitpick:  just to make it less confusing
```suggestion
		log.Info("CSI driver provisioner not needed")
```
The server is still needed to mount host monitoring pods 🙃